src/ERC7540Vault.sol

```diff
@@ -34,6 +34,7 @@ contract ERC7540Vault is Auth, IERC7540 {
     /// @notice The restricted ERC-20 vault share (tranche token).
     ///         Has a ratio (token price) of underlying assets exchanged on deposit/mint/withdraw/redeem.
     address public immutable share;
+    uint8 public immutable shareDecimals;
```

Comment:
I'm wondering whether we should also store `assetDecimals`, and then always pass `assetDecimals` + `shareDecimals` on every call to `InvestmentManager`. This would save a lot of storage lookups since almost all methods in `InvestmentManager` need to know these decimals.

TBD, let me know what you think!

Reply:
I cant think of any reason not to!